src/index.js

```diff
@@ -0,0 +1,40 @@
+"use strict";
+import { Minimatch } from "minimatch";
+import { dirname, basename } from "path";
+
+export const renamer = (options) => {
+  return (files, _, done) => {
+    Object.keys(options).map((opt) => {
+      const matcher = Minimatch(options[opt].pattern);
+
+      Object.keys(files).map((file) => {
+        if (!matcher.match(file)) {
+          return;
+        }
+
+        const nameTransformer = options[opt].rename;
+        let newFilename = dirname(file);
+
+        // If file is at root of the `source` directory, strip the relative file path
+        if (newFilename === ".") {
+          newFilename = "";
+        }
+
+        if (typeof nameTransformer === "function") {
+          newFilename += nameTransformer(basename(file));
```

Comment:
Concatenation of dirname and basename loses the path separator (no matter which one it is). I think you want something like:

```javascript
newFilename = path.join(newFilename, nameTransformer(basename(file)));
```